### PR TITLE
Add rpc interface for tiflash disaggregated architecture

### PR DESCRIPTION
**Pull request by JaySon-Huang**

### CODE REVIEW

File: proto/disaggregated.proto [hunk 1]
```diff
@@ -43,3 +47,73 @@ message TryMarkDeleteRequest {
...
+    int64 task_id = 5; // if task id is -1 , it indicates a tidb task.
+    string executor_id = 6; // the exectuor id
+}
+message DisaggReadError {
+	int32 code = 1;
+	string msg = 2;
+}
+
+message EstablishDisaggregatedTaskRequest {
+    DisaggregatedTaskMeta meta = 1;
```
Comment (breezewish) [inline proto/disaggregated.proto:68]: How about Let's unify the name as Disagg?

File: proto/disaggregated.proto [hunk 2]
```diff
@@ -43,3 +47,73 @@ message TryMarkDeleteRequest {
...
+    string address = 2; // target address of this task.
+    int64 timeout = 3;
```
Comment (breezewish) [inline proto/disaggregated.proto:70]: Add unit (e.g. ms / s)?

File: proto/disaggregated.proto [hunk 3]
```diff
@@ -43,3 +47,73 @@ message TryMarkDeleteRequest {
...
+message EstablishDisaggregatedTaskResponse {
+    DisaggReadError error = 1;
+    // The read node needs to update its region cache about
+    // these regions
+    repeated metapb.Region retry_regions = 2;
+    // Write node maintains a snapshot with a lease time.
+    // Read node should read the delta pages
+    // (ColumnFileInMemory and ColumnFileTiny)
+    // along with this store_id and snapshot_id.
+    uint64 store_id = 3; // metapb.Store.id
```
Comment (breezewish) [inline proto/disaggregated.proto:90]: Where is snapshot_id?

File: proto/disaggregated.proto [hunk 4]
```diff
@@ -43,3 +47,73 @@ message TryMarkDeleteRequest {
...
+}
+
+message FetchDisaggregatedPagesRequest {
+    // The snapshot id to fetch pages 
+    DisaggregatedTaskMeta meta = 1;
+    int64 table_id = 2;
+    uint64 segment_id = 3;
+    // It must be a subset of the delta_pages returned
+    // in EstablishDisaggregatedTaskResponse.segments
+    repeated uint64 pages = 4;
```
Comment (breezewish) [inline proto/disaggregated.proto:102]: page_ids?

File: proto/disaggregated.proto [hunk 5]
```diff
@@ -43,3 +47,73 @@ message TryMarkDeleteRequest {
...
+    // target address of this task
+    string address = 5;
```
Comment (breezewish) [inline proto/disaggregated.proto:104]: Not sure, but why is the address embedded in the request?

**Review by breezewish — APPROVED** [review]
Most looks good

File: proto/disaggregated.proto (hunk 3 above)
Comment (JaySon-Huang) [reply proto/disaggregated.proto:92]: add in the below line, we use `DisaggTaskMeta` as snapshot_id now

File: proto/disaggregated.proto (hunk 2 above)
Comment (JaySon-Huang) [reply proto/disaggregated.proto]: added

File: proto/disaggregated.proto (hunk 1 above)
Comment (JaySon-Huang) [reply proto/disaggregated.proto]: done

File: proto/disaggregated.proto (hunk 4 above)
Comment (JaySon-Huang) [reply proto/disaggregated.proto]: done

File: proto/disaggregated.proto (hunk 5 above)
Comment (JaySon-Huang) [reply proto/disaggregated.proto]: Removed. We can get the address without defining it in the proto

**Review by breezewish — APPROVED** [review]

File: proto/tikvpb.proto [hunk 6]
```diff
@@ -121,6 +121,10 @@ service Tikv {
     rpc tryAddLock(disaggregated.TryAddLockRequest) returns (disaggregated.TryAddLockResponse) {};
     /// Try to delete a S3 object, atomically
     rpc tryMarkDelete(disaggregated.TryMarkDeleteRequest) returns (disaggregated.TryMarkDeleteResponse) {};
+    /// Build the disaggregated task on TiFlash write node
+    rpc EstablishDisaggregatedTask(disaggregated.EstablishDisaggTaskRequest) returns (disaggregated.EstablishDisaggTaskResponse) {}
```
Comment (breezewish) [inline proto/tikvpb.proto:125]:
```suggestion
    rpc EstablishDisaggTask(disaggregated.EstablishDisaggTaskRequest) returns (disaggregated.EstablishDisaggTaskResponse) {}
```

File: proto/tikvpb.proto [hunk 7]
```diff
@@ -121,6 +121,10 @@ service Tikv {
...
+    /// Exchange page data between TiFlash write node and compute node
+    rpc FetchDisaggregatedPages(disaggregated.FetchDisaggPagesRequest) returns (stream disaggregated.PagesPacket) {}
```
Comment (breezewish) [inline proto/tikvpb.proto:127]:
```suggestion
    rpc FetchDisaggPages(disaggregated.FetchDisaggPagesRequest) returns (stream disaggregated.PagesPacket) {}
```

Comment (JaySon-Huang) [reply proto/tikvpb.proto]: renamed

File: proto/tikvpb.proto (hunk 6 above)
Comment (JaySon-Huang) [reply proto/tikvpb.proto]: renamed

**Pull request merged**